i3_resurrect/main.py

```diff
@@ -98,21 +98,19 @@ def restore_workspace(workspace, numeric, directory, profile, target):
     if profile is not None:
         directory = Path(directory) / 'profiles'
 
-    # Switch to the workspace which we are loading.
-    if numeric:
-        if workspace.isdigit():
-            i3.command(
-                f'workspace --no-auto-back-and-forth number {workspace}'
-            )
-        else:
-            util.eprint('Invalid workspace number.')
-            sys.exit(1)
+    # Get layout name from file
+    workspace_layout = layout.read(workspace, directory, profile)
+    if 'name' in workspace_layout:
+        workspace_name = workspace_layout["name"]
     else:
-        i3.command(f'workspace --no-auto-back-and-forth {workspace}')
+        workspace_name = workspace
+
+    # Switch to the workspace which we are loading.
+    i3.command(f'workspace --no-auto-back-and-forth {workspace_name}')
 
     if target != 'programs_only':
         # Load workspace layout.
-        layout.restore(workspace, numeric, directory, profile)
+        layout.restore(workspace, numeric, profile, workspace_layout)
```

Comment:
Shouldn't you be passing `workspace_name` instead of `workspace` here? Otherwise in `layout.restore()` it may get the windows list from a different workspace than the one you are on.

Reply:
Yes true

Reply:
I Have to set numeric to False also

Reply:
Think is ok now but may be need some adjustments. Also, the use of workspace and workspace_name  can be title confusing.